service/matching/task_reader.go

```diff
@@ -83,30 +79,8 @@ func newTaskReader(backlogMgr *backlogManagerImpl) *taskReader {
 
 // Start taskReader background goroutines.
 func (tr *taskReader) Start() {
-	if !atomic.CompareAndSwapInt32(
-		&tr.status,
-		common.DaemonStatusInitialized,
-		common.DaemonStatusStarted,
-	) {
-		return
-	}
```

Comment:
I came to the conclusion that this isn't needed since the only place this is started from is the `backlogManagerImpl` which in turn is started by `physicalTaskQueueManagerImpl` - which has this protection already.

Reply:
We don't need this status cas stuff anywhere and I'm slowly ripping it out. It's protection against calling Start/Stop multiple times and we should just never do that, and treat it as a bug if it somehow happens. For components with the same lifecycle as the process, fx takes care of everything. For things like this, matching engine should manage the lifecycle accurately and never start/stop twice.